### PR TITLE
Add support for gorilla pages for tier 0.

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

Introduce a new page type that uses gorilla buffers for compressing live data.

- The gorilla implementation lives under `libnetdata/gorilla`, along with fuzzy tests and benchmarks.
- The `database/page.c` file contains the main logic when it comes to handling gorilla buffer allocation, deallocation and flush.
- The data file page descriptor for gorilla pages had to be modified in order to encode the number of entries and the end time as a delta from the start time.
- The buffers of gorilla pages are serialized together. When deserializing page we perform one allocation and patch the pointers to each buffer.
- Overall, compression ratio is around ~75%. Two new charts where added under `Netadata monitoring`. One of them shows the number of gorilla buffers we allocate per second, and the other shows the overall savings prior to sending data for compression.

NOTE: Collection with gorilla pages is disabled for the time being, ie. the agent can only read such pages, not create them.

##### Test Plan

This PR has been running on parent 3 for a couple weeks now. The intention is to merge this without changing the default page type.

### CODE REVIEW

File: libnetdata/log/log.c [hunk 1]
```diff
@@ -606,12 +606,12 @@ void open_all_log_files() {
...
-    open_log_file(STDERR_FILENO, stderr, stdcollector_filename, &collector_log_syslog, 0, NULL);
+    // open_log_file(STDERR_FILENO, stderr, stdcollector_filename, &collector_log_syslog, 0, NULL);
 
     // Netdata starts using stderr and if it has success to open file it redirects
-    FILE *fp = open_log_file(stdcollector_fd, NULL, stderr_filename, &error_log_syslog, 1, &stdcollector_fd);
-    if (fp)
-        stderror = fp;
+    // FILE *fp = open_log_file(stdcollector_fd, NULL, stderr_filename, &error_log_syslog, 1, &stdcollector_fd);
+    // if (fp)
+        // stderror = fp;
```
Comment (stelfrag) [inline libnetdata/log/log.c:615]: Are these commented out for debugging?

File: libnetdata/log/log.c [hunk 2]
```diff
@@ -627,7 +627,7 @@ void open_all_log_files() {
 // error log throttling
 
 time_t error_log_throttle_period = 1200;
-unsigned long error_log_errors_per_period = 200;
+unsigned long error_log_errors_per_period = 999999;
```
Comment (stelfrag) [inline libnetdata/log/log.c:630]: Was this change for debugging?

File: libnetdata/log/log.c (hunk 1 above)
Comment (vkalintiris) [reply libnetdata/log/log.c]: +1. Removed.

**Comment by thiagoftsm** [comment]
Hello @vkalintiris ,

Please, can you rebase your PR?

Best regards!

**Comment by vkalintiris** [comment]
@stelfrag AFAICT, reading/writing gorilla pages works correctly on 32-bit system. I tested the PR on a i386 debian image under qemu-system emulation.

**Review by stelfrag — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected with different memory modes. LGTM!

**Pull request merged**